ydb/core/tx/datashard/datashard.cpp

```diff
@@ -3054,7 +3054,11 @@ bool TDataShard::CheckDataTxRejectAndReply(const NEvents::TDataEvents::TEvWrite:
                 status = NKikimrDataEvents::TEvWriteResult::STATUS_OVERLOADED;
                 break;
             case NKikimrTxDataShard::TEvProposeTransactionResult::ERROR:
-                status = NKikimrDataEvents::TEvWriteResult::STATUS_INTERNAL_ERROR;
+                if (rejectReasons == ERejectReasons::WrongState) {
```

Comment:
(rejectReasons | ERejectReasons::WrongState) != 0

rejectReasons is a set of flags